### DISCUSSION TITLE
December 24th TODO

### DISCUSSION THREAD

**Discussion by TheATeam-566** [TheATeam-566/Hermit#69] [Standup]
## 👋 December 24th TODO ✏️
See #65, our current big blocker.

> Chris and I are now able to access Google sign in via backend localhost:8000/auth/google.
>
> The challenge we are having now, is that after signing via Google sign in, we need to provide a User object to the callback. We currently do not have a User object defined in the Firestore database, therefore we can not provide the callback with a User object.
>
> This brings up a few major challenges for us to solve in the immediate future. We should focus on designing our NoSQL database to handle (Users, UserOrderHistory, etc.).
>
> **Roadblock** for this particular issue? Retrieving User data from our Firestore Database to be provided to the Google sign-in callback.

We need to discuss the following:
- What the Firestore user schema (and user order history?) will look like
- How we're going to persist user account w/ authentication received from Google Oauth2.0 profile scope
- can we config concurrently? I think it's hiding messages from us or something

💪💪💪

**Comment by chrispinkney** [comment]
PS Merry Christmas

**Reply by chrispinkney** [reply]
🎄 🎅

**Comment by chrispinkney** [comment]
Oh, and btw we need to talk about cookies. 🍪 

Something, something [cookie session](https://github.com/expressjs/cookie-session)?
